recipes/libcurl/all/conanfile.py

```diff
@@ -296,14 +296,38 @@ def _patch_autotools(self):
     def _patch_cmake(self):
         if not self._is_using_cmake_build:
             return
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
         # Custom findZstd.cmake file relies on pkg-config file, make sure that it's consumed on all platforms
         if self._has_zstd_option:
             tools.replace_in_file(os.path.join(self._source_subfolder, "CMake", "FindZstd.cmake"),
                                   "if(UNIX)", "if(TRUE)")
         # TODO: check this patch, it's suspicious
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+        tools.replace_in_file(cmakelists,
                               "include(CurlSymbolHiding)", "")
 
+        # libnghttp2
+        tools.replace_in_file(
+            cmakelists,
+            "find_package(NGHTTP2 REQUIRED)",
+            "find_package(libnghttp2 REQUIRED CONFIG)",
+        )
+        tools.replace_in_file(
+            cmakelists,
+            "include_directories(${NGHTTP2_INCLUDE_DIRS})",
+            "",
+        )
+        tools.replace_in_file(
+            cmakelists,
+            "list(APPEND CURL_LIBS ${NGHTTP2_LIBRARIES})",
+            "list(APPEND CURL_LIBS libnghttp2::nghttp2)",
+        )
+
+        tools.replace_in_file(
+            cmakelists,
+            "find_package(OpenSSL REQUIRED)",
+            "find_package(OpenSSL REQUIRED CONFIG)",
+        )
```

Comment:
why not add `cmake_find_package` generator instead?

Reply:
@SpaceIm I fixed this, as I noticed, that the cmake provided FindOpenSSL.cmake was used. During my tests I also tried the cmake_find_package generator. This works as well. What is your preferd way?

Reply:
Adding `cmake_find_package` is better than a patch.